app/scheduled/circulars/uploadTar.ts

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3'
+import { createReadableStreamFromReadable } from '@remix-run/node'
+import { Readable } from 'stream'
+import type { Pack } from 'tar-stream'
+import { pack as tarPack } from 'tar-stream'
+
+import { getEnvOrDie } from '~/lib/env.server'
+import type { Circular } from '~/routes/circulars/circulars.lib'
+import { formatCircular } from '~/routes/circulars/circulars.lib'
+
+interface TarContext {
+  context: {
+    pack: Pack
+    tarStream: Readable | ReadableStream<Uint8Array>
+    fileType: string
+  }
+}
```

Comment:
Why do you have an extra object nested inside? Why not the following:

```suggestion
interface TarContext {
  pack: Pack
  tarStream: Readable | ReadableStream<Uint8Array>
  fileType: string
}
```

Reply:
I think that's a holdover from when I was passing data back in the action. it had to match the context object shape. I'll adjust it.